api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -199,57 +214,60 @@ public void setResearchPurposeApproved(String ns, String firecloudName, boolean
     saveWithLastModified(workspace, now);
   }
 
-  @Override
-  public Workspace updateUserRoles(Workspace workspace, Set<WorkspaceUserRole> userRoleSet) {
-    Map<Long, WorkspaceUserRole> userRoleMap = new HashMap<Long, WorkspaceUserRole>();
-    for (WorkspaceUserRole userRole : userRoleSet) {
-      userRole.setWorkspace(workspace);
-      userRoleMap.put(userRole.getUser().getUserId(), userRole);
+  private void updateFirecloudAclsOnUser(
```

Comment:
This method doing mutation feels a bit awkward, I would probably restructure this as:

```
static WorkspaceACLUpdate accessLevelToWorkspaceACLUpdate(accessLevel)
```

Optionally include user email there, or just add it after.